tests/Unit/DataStructures/Test_Mesh.cpp

```diff
@@ -14,99 +14,126 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Mesh", "[DataStructures][Unit]") {
   SECTION("Uniform LGL mesh") {
     const Mesh<1> mesh1d_lgl{3, Spectral::Basis::Legendre,
                              Spectral::Quadrature::GaussLobatto};
-    CHECK(mesh1d_lgl[0] == 3);
+    CHECK(mesh1d_lgl.extents() == Index<1>{3});
     CHECK(mesh1d_lgl.number_of_grid_points() == 3);
-    CHECK(mesh1d_lgl.bases()[0] == Spectral::Basis::Legendre);
-    CHECK(mesh1d_lgl.quadratures()[0] == Spectral::Quadrature::GaussLobatto);
+    CHECK(mesh1d_lgl.basis(0) == Spectral::Basis::Legendre);
+    CHECK(mesh1d_lgl.quadrature(0) == Spectral::Quadrature::GaussLobatto);
     const Mesh<2> mesh2d_lgl{3, Spectral::Basis::Legendre,
                              Spectral::Quadrature::GaussLobatto};
-    CHECK(mesh2d_lgl[0] == 3);
-    CHECK(mesh2d_lgl[1] == 3);
+    CHECK(mesh2d_lgl.extents() == Index<2>{3, 3});
     CHECK(mesh2d_lgl.number_of_grid_points() == 9);
-    CHECK(mesh2d_lgl.bases()[0] == Spectral::Basis::Legendre);
-    CHECK(mesh2d_lgl.quadratures()[0] == Spectral::Quadrature::GaussLobatto);
-    CHECK(mesh2d_lgl.bases()[1] == Spectral::Basis::Legendre);
-    CHECK(mesh2d_lgl.quadratures()[1] == Spectral::Quadrature::GaussLobatto);
+    CHECK(mesh2d_lgl.basis(0) == Spectral::Basis::Legendre);
+    CHECK(mesh2d_lgl.basis(1) == Spectral::Basis::Legendre);
+    CHECK(mesh2d_lgl.quadrature(0) == Spectral::Quadrature::GaussLobatto);
+    CHECK(mesh2d_lgl.quadrature(1) == Spectral::Quadrature::GaussLobatto);
```

Comment:
check both the array access and individual access for all three of basis quadrature and extents